modules/peridynamics/src/auxkernels/NodalRankTwoPD.C

```diff
@@ -85,23 +80,42 @@ NodalRankTwoPD::NodalRankTwoPD(const InputParameters & parameters)
   for (unsigned int i = 0; i < coupledComponents("displacements"); ++i)
     _disp_var.push_back(getVar("displacements", i));
 
-  if (_rank_two_tensor == "stress" && !isParamValid("youngs_modulus") &&
-      !isParamValid("poissons_ratio"))
-    mooseError("Both Young's modulus and Poisson's ratio must be provided for stress calculation");
+  _Cijkl.zero();
+  if (_rank_two_tensor == "stress")
+  {
+    if (!isParamValid("youngs_modulus") || !isParamValid("poissons_ratio"))
+      mooseError(
+          "Both Young's modulus and Poisson's ratio must be provided for stress calculation");
+    else
```

Comment:
You don't actually need to put the following lines of code inside the `else` block, but it doesn't hurt anything, so I'll let that slide.